### ISSUE TITLE
BuildReducedGraphVisitor::visit_invoc panic on unwrap

### ISSUE THREAD

**Issue by Kerollmops** [rust-lang/rust#36912]
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:323

I tried to compile the [objc crate](https://github.com/SSheldon/rust-objc), [the `unwrap` failing seems to be here](https://github.com/rust-lang/rust/blob/634ecf038d477b8e9aeb815d4cca64939c05781c/src/librustc_resolve/build_reduced_graph.rs#L534)

I expected to see this happen: The compilation working.

Instead, this happened: The compilation making the compiler to crash.

## Meta

`rustc --version --verbose`:
```
rustc 1.14.0-nightly (289f3a4ca 2016-09-29)
binary: rustc
commit-hash: 289f3a4ca79916d6445b452fc19a18a1e42a879a
commit-date: 2016-09-29
host: x86_64-apple-darwin
release: 1.14.0-nightly
```

Backtrace:
```
   1:        0x109167938 - std::sys::backtrace::tracing::imp::write::h22f199c1dbb72ba2
   2:        0x1091750af - std::panicking::default_hook::{{closure}}::h9a389c462b6a22dd
   3:        0x1091725ad - std::panicking::default_hook::h852b4223c1c00c59
   4:        0x109172c46 - std::panicking::rust_panic_with_hook::hcd9d05f53fa0dafc
   5:        0x109172ae4 - std::panicking::begin_panic::hf6c488cee66e7f17
   6:        0x109172a02 - std::panicking::begin_panic_fmt::hb0a7126ee57cdd27
   7:        0x109172967 - rust_begin_unwind
   8:        0x1091b2c20 - core::panicking::panic_fmt::h9af671b78898cdba
   9:        0x1091b2b24 - core::panicking::panic::h1a2d1a6b50eaa468
  10:        0x105717033 - rustc_resolve::build_reduced_graph::BuildReducedGraphVisitor::visit_invoc::h1ffde3beaeda5557
  11:        0x10570f922 - rustc_resolve::macros::<impl syntax::ext::base::Resolver for rustc_resolve::Resolver<'a>>::visit_expansion::ha1f7cee58f82f9da
  12:        0x108d9e5eb - syntax::ext::expand::MacroExpander::collect_invocations::hac509c4490fb88f2
  13:        0x108d9c3a2 - syntax::ext::expand::MacroExpander::expand::h737d7f1635448d5d
  14:        0x108d9adc5 - syntax::ext::expand::MacroExpander::expand_crate::hd89d06b7a6333976
  15:        0x108da552d - syntax::ext::expand::expand_crate::h80012253cfa7f1f5
  16:        0x10502023b - rustc_driver::driver::phase_2_configure_and_expand::{{closure}}::hc795707fb140075e
  17:        0x104fd8844 - rustc_driver::driver::phase_2_configure_and_expand::h04526112c643cb41
  18:        0x104fd3222 - rustc_driver::driver::compile_input::h5b63ccd49eeeb98b
  19:        0x104ffb579 - rustc_driver::run_compiler::h98c7274e7cb1d11d
  20:        0x104f3ce98 - std::panicking::try::do_call::h99ed0da044e497c3
  21:        0x10917566a - __rust_maybe_catch_panic
  22:        0x104f5c6df - <F as alloc::boxed::FnBox<A>>::call_box::hbdd5a14cd8e33b97
  23:        0x109171864 - std::sys::thread::Thread::new::thread_start::h50b05608a499d2b2
  24:     0x7fff99642aaa - _pthread_body
  25:     0x7fff996429f6 - _pthread_start
```

**Comment by jonas-schievink** [comment]
Duplicate of #36816, fixed in #36819

**Comment by Kerollmops** [comment]
Ooops, sorry !
